lib/serializer.ts

```diff
@@ -512,13 +558,46 @@ export class Serializer {
    * @returns {object|string|Array|number|boolean|Date|stream} A valid deserialized Javascript object
    */
   deserialize(mapper: Mapper, responseBody: any, objectName: string): any {
-    if (responseBody === null || responseBody === undefined) return responseBody;
+    if (responseBody == undefined) {
+      if (this.isXML && mapper.type.name === "Sequence" && !mapper.xmlIsWrapped) {
+        // Edge case for empty XML non-wrapped lists. xml2js can't distinguish
+        // between the list being empty versus being missing,
+        // so let's do the more user-friendly thing and return an empty list.
+        responseBody = [];
+      } else {
+        return responseBody;
+      }
```

Comment:
Isn't the else part redundant ? The outer return responseBody will eventually happen any ways.

Reply:
I think Dan brought this up, see https://github.com/Azure/ms-rest-js/pull/19#discussion_r178192141